src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverter.cs

```diff
@@ -62,7 +62,8 @@ protected override void CreateCollection(ref Utf8JsonReader reader, ref ReadStac
                     return false;
                 }
 
-                if (!converter.TryWrite(writer, enumerator.Current, options, ref state))
+                object? element = enumerator.Current;
```

Comment:
Note this is for consistency with other code. Also applies to another case of this later.